pkg/client/fake/client.go

```diff
@@ -1080,7 +1082,24 @@ type fakeSubResourceClient struct {
 }
 
 func (sw *fakeSubResourceClient) Get(ctx context.Context, obj, subResource client.Object, opts ...client.SubResourceGetOption) error {
-	panic("fakeSubResourceClient does not support get")
+	switch sw.subResource {
+	case "scale":
+		scale, isScale := subResource.(*autoscalingv1.Scale)
+		if !isScale {
+			return apierrors.NewBadRequest(fmt.Sprintf("got invalid type %t, expected Scale", subResource))
+		}
+		if err := sw.client.Get(ctx, client.ObjectKeyFromObject(obj), obj); err != nil {
+			return err
+		}
+		scaleOut, err := extractScale(obj)
+		if err != nil {
+			return err
+		}
+		*scale = scaleOut
+		return nil
+	default:
+		return fmt.Errorf("fakeSubResourceClient does not support get for %s", sw.subResource)
+	}
```

Comment:
Having to use the subresource client to then use the actual client seems like we wouldn't need the sub resource client to do the GET.  The value of the scale is indicative on the spec and the subresource client has to get that value from the actual client where it's derived.

Reply:
This is implemented on the server-side as far as I know. I cannot just lookup on the client because the fake client doesn't know how to extract this information. It's something of a meta-resource.

Reply:
Fair and not criticizing the implementation but asking because you need to deal with this the way you are dealing with it.